spec/facts_schema_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'digest'
+require 'pathname'
+require 'json_schemer'
+
+schema = {
```

Comment:
Did you ask the facter devs if they provide any schema?

Reply:
I am not really concerned about the validity of the schema overall, since most of them were generated by Facter (hopefully not modified by hand). I am looking for a way to enforce other changes that wouldn't be part of a facter json schema, for example, IP address always being the same, blanked out UUID values, etc. that would help make creating tests with factsets easier.

Reply:
ah yes, makes sense. I also asked for a schema, so we can properly validate factsets: https://github.com/puppetlabs/facter/issues/2730